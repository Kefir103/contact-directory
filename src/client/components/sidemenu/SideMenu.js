import React from 'react';
import LoadBtnContainer from './LoadBtnContainer';
import { connect } from 'react-redux';
import FilterContainer from './FilterContainer';

const SideMenu = () => {
    return (
        <aside>
            <LoadBtnContainer />
            <FilterContainer />
        </aside>
    );
};

export default connect()(SideMenu);
