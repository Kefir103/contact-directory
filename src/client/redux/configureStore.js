import { createStore, applyMiddleware, compose } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension/index';
import { rootReducer } from './reducers/rootReducer';
import thunk from 'redux-thunk';

let store;

export default function configureStore(initialState) {
    if (store) {
        return store;
    }

    store =
        process.env.NODE_ENV === 'production'
            ? createStore(rootReducer, initialState, compose(applyMiddleware(thunk)))
            : createStore(rootReducer, initialState, composeWithDevTools(applyMiddleware(thunk)));

    return store;
}
