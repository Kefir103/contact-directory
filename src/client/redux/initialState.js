export const initialState = {
    filter: {
        sortingMap: new Map([]),
        filterText: '',
        filterFields: [],
        currentPage: 1,
    },
    data: {
        currentElement: {
            elementInfo: undefined,
            elementIndex: -1,
        },
        elements: [],
        currentElements: [],
        filteredElements: undefined,
        inputContainer: {
            isAddButtonDisabled: false,
            isFormOpen: false,
            inputElement: {
                id: 0,
                firstName: '',
                lastName: '',
                email: '',
                phone: '',
            },
            validInputs: {
                id: false,
                firstName: false,
                lastName: false,
                email: false,
                phone: false,
            },
            addingStatus: {
                addingStatusText: '',
                isError: false,
            },
        },
    },
    appStatus: {
        isLoading: false,
        error: undefined,
    },
};
