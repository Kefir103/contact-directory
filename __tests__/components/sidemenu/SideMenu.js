import React from 'react';
import { shallow, render } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import SideMenu from '../../../src/client/components/sidemenu/SideMenu';
import { initialState } from '../../../src/client/redux/initialState';

const mockStore = configureMockStore();

describe('<SideMenu />', () => {
    test('should render correctly with empty fullData', () => {
        const store = mockStore({ ...initialState });
        const wrapper = shallow(<SideMenu store={store} />);
        const component = wrapper.dive().dive();
        expect(component).toMatchSnapshot();
    });
    test('should render correctly with filled fullData', () => {
        const store = mockStore({
            ...initialState,
            data: {
                ...initialState.data,
                fullData: [1, 2, 3],
            },
        });
        const wrapper = shallow(<SideMenu store={store} />);
        const component = wrapper.dive().dive();
        expect(component).toMatchSnapshot();
    });
});
