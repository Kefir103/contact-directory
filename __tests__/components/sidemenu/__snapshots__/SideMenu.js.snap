// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideMenu /> should render correctly with empty fullData 1`] = `
<aside>
  <Connect(LoadBtnContainer) />
</aside>
`;

exports[`<SideMenu /> should render correctly with filled fullData 1`] = `
<aside>
  <Connect(LoadBtnContainer) />
  <Connect(FilterContainer) />
</aside>
`;
